### ISSUE TITLE
The `name` argument of client.images.list not respected with engine 20.10/API 1.41

### ISSUE THREAD

**Issue by william-richard** [docker/docker-py#2722]
With previous version of the docker engine (19.03 and earlier), if you called `client.images.list(<repository_name>)`, it would only return images that are part of the provided `repository_name`.  With docker engine 20.10, the same call now returns all images.

example code:
```
import docker

if __name__ == "__main__":
    docker_client = docker.from_env()
    
    # start with no images
    docker_client.images.prune(filters={'dangling':False})
    # this prints the dind image, since I'm running OSX
    print(docker_client.images.list())

    # pull some images
    docker_client.images.pull('python')
    docker_client.images.pull('hello-world')

    # according to the docs and with previous version of the docker engine,
    # this should only print the python image.
    # But with engine 20.3 it also lists the hello-world and dind images
    print(docker_client.images.list("python"))
```

Expected result (docker engine 19.03):
```
[<Image: 'docker:19.03.13-dind'>]
[<Image: 'python:latest'>]
```

<details><summary>19.03 test version info</summary>
<p>

```
$ poetry show | grep docker && python --version && docker version
docker           4.4.0     A Python library for the Docker Engine API.
Python 3.9.0
Client: Docker Engine - Community
 Cloud integration: 1.0.2
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        4484c46d9d
 Built:             Wed Sep 16 16:58:31 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.13
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       4484c46d9d
  Built:            Wed Sep 16 17:07:04 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.3.7
  GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

</p>
</details>

Actual result (docker engine 20.10):
```
[<Image: 'docker:19.03.13-dind'>]
[<Image: 'python:latest'>, <Image: 'docker:19.03.13-dind'>, <Image: 'hello-world:latest'>]
```

Version info for the 20.10 test:
```
$ poetry show | grep docker && python --version && docker version
docker           4.4.0     A Python library for the Docker Engine API.
Python 3.9.0
Client: Docker Engine - Community
 Cloud integration: 1.0.4
 Version:           20.10.0
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        7287ab3
 Built:             Tue Dec  8 18:55:43 2020
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.0
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       eeddea2
  Built:            Tue Dec  8 18:58:04 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by JonZeolla** [comment]
I'm seeing the same.
```bash
$ pip3 list | grep docker
docker                   4.4.0
```
```bash
$ docker version
Client: Docker Engine - Community
 Cloud integration: 1.0.7
 Version:           20.10.2
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        2291f61
 Built:             Mon Dec 28 16:12:42 2020
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.2
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       8891c58
  Built:            Mon Dec 28 16:15:28 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by william-richard** [comment]
It appears this problem was fixed by https://github.com/docker/docker-py/pull/2708, which was released in version 4.4.1 (but was not mentioned in the changelog 😿 )
